src/main/java/unicash/model/category/UniqueCategoryList.java

```diff
@@ -199,4 +200,34 @@ public static boolean categoriesAreUnique(List<Category> categories) {
         }
         return true;
     }
+
+    /**
+     * Returns all categories as a single, unified string.
+     *
+     * @return all categories as a string
+     */
+    public String joinCategoriesAsString() {
+        String categoriesString = internalList
+                .stream()
+                .map(category -> category.toString())
+                .collect(Collectors.joining(","));
+
+        requireNonNull(categoriesString);
```

Comment:
`collect` should return an empty string if no entries, so maybe check for that instead?

Reply:
Fixed

Reply:
Scratch that, for this and the previous comment, I've removed both `requireNonNull` and the above String checking. The reason is because categoriesString and categoryList are intended to return an empty list and empty string if a Transaction does not have any categories assigned. Checking if a particular transaction is empty should be done on the callling method's side. In light of that, I have added the `UniqueCategoryList#isEmpty()` method.